hazelcast/src/hazelcast/client/sql.cpp

```diff
@@ -58,7 +70,21 @@ sql_service::execute(const sql_statement& statement)
 {
     using protocol::ClientMessage;
 
-    auto query_conn = query_connection();
+    auto statement_par_arg_index_ptr = statement.partition_argument_index();
+    int32_t statement_par_arg_index = statement_par_arg_index_ptr != nullptr
+                                        ? statement_par_arg_index_ptr->load()
+                                        : -1;
+
+    auto arg_index = statement_par_arg_index != -1
+                       ? statement_par_arg_index
+                       : *(partition_argument_index_cache_->get_or_default(
+                           statement.sql(), std::make_shared<int32_t>(-1)));
+
+    auto partition_id = extract_partition_id(statement, arg_index);
+    std::shared_ptr<connection::Connection> query_conn =
+      partition_id != boost::none ? query_connection(partition_id.value())
```

Comment:
partition_id ? query_connection(partition_id.value())

Reply:
Fixed. As this PR is merged, the changes are made in another branch. https://github.com/hazelcast/hazelcast-cpp-client/pull/1188